rna/analytics.py

```diff
@@ -0,0 +1,245 @@
+    """
+    for the target_class, a vector of length n_single_cell_types with 1 or more 1's, give
+    back the columns in the mixtures that contain one or more of these single cell types
+
+    :param target_class: vector of length n_single_cell_types with at least one 1
+    :param priors: vector of length n_single_cell_types with 0 or 1 to indicate single cell type has 0 or 1 prior,
+    uniform assumed otherwise
+    :return: list of ints, in [0, 2 ** n_cell_types]
+    """
+
+    def int_to_binary(i):
+        binary = bin(i)[2:]
+        while len(binary) < len(single_cell_types):
+            binary = '0' + binary
+        return np.flip([int(j) for j in binary]).tolist()
+
+    def binary_admissable(binary, target_class, priors):
+        """
+        gives back whether the binary (string of 0 and 1 of length n_single_cell_types) has at least one of
+        target_class in it, and all priors satisfied
+        """
+        if priors:
+            for i in range(len(target_class)):
+                # if prior is zero, the class should not occur
+                if binary[i] == 1 and priors[i] == 0:
+                    return False
+                # if prior is one, the class should occur
+                # as binary is zero it does not occur and return False
+                if binary[i] == 0 and priors[i] == 1:
+                    return False
+        # at least one of the target class should occur
+        if np.inner(binary, target_class)==0:
+            return False
+        return True
+
+    return [i for i in range(2 ** len(single_cell_types)) if binary_admissable(int_to_binary(i), target_class, priors)]
+
+
+# def classify_single(X, y_nhot):
+#     """
+#     Very simple analysis of single cell type classification, useful as
+#     preliminary test.
+#     """
+#     # classify single classes
+#     single_samples = combine_samples(X)
+#     y = from_nhot_to_labels(y_nhot)
+#     print('fitting on {} samples, {} features, {} classes'.format(
+#         X.shape[0],
+#         single_samples.shape[1],
+#         len(set(y)))
+#     )
+#
+#     X_train, X_test, y_train, y_test = train_test_split(single_samples, y, stratify=y)
+#     single_model = MLPClassifier(random_state=0, max_iter=1000)
+#     single_model.fit(X_train, y_train)
+#     y_pred = single_model.predict(X_test)
+#     print('train accuracy for single classes: {}'.format(
+#         accuracy_score(y_test, y_pred))
+#     )
+#
+#     plt.plot(single_model.loss_curve_)
+#     plt.xlabel("Steps")
+#     plt.ylabel("Loss")
+#     plt.show()
+#
+#     # Compute confusion matrix
+#     cnf_matrix = confusion_matrix(y_test, y_pred)
+#     np.set_printoptions(precision=2)
+#     # print(cnf_matrix)
+
+
+def remove_markers(X):
+    """
+    Removes the gender and control markers.
+    """
+    try:
+        X = X[:, :-4]
+    except IndexError:
+        X = np.array([X[i][:, :-4] for i in range(X.shape[0])])
+
+    return X
+
+
+def cllr(lrs, y_nhot, target_class):
```

Comment:
zit dit niet in lir?

Reply:
Ja die gebruik ik, maar had nog niet de juiste input argumenten daarvoor en die 'maak' ik in deze functie.